server/pulp/server/db/manage.py

```diff
@@ -165,7 +165,16 @@ def ensure_database_indexes():
 
     # Load all the model classes that the server knows about and ensure their indexes as well
     plugin_manager = PluginManager()
-    for model_class in plugin_manager.unit_models.itervalues():
+    for unit_type, model_class in plugin_manager.unit_models.items():
+        unit_key_index = {'fields': model_class.unit_key_fields, 'unique': True}
+        for index in model_class._meta['indexes']:
+            if unit_key_index == index:
```

Comment:
dict equality should be `sorted(dict1.keys()) == sorted(dict2.keys()) and sorted(dict1.values()) == sorted(dict2.values())`[0], at least as far back as python 2.4[1], but a unit test proving it would be ideal.

(look for the mappings bullet point in both of the referenced links)
[0]: https://docs.python.org/2/reference/expressions.html#not-in
[1]: https://docs.python.org/2.4/ref/comparisons.html